From 58b8806d87327a21f3d5376fd9dd626d75f5886f Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 28 Jan 2019 18:07:07 +0100
Subject: [PATCH 0921/1051] arm64: mvebu: cn9130: fix and simplify pcie
 description

- the PCIe mem size should be reduced since last 1MB is reserved for
configuration space. The CP110_PCIE_MEM_SIZE is used for defining the base
address of config space.

- remove redundant pcie description from cn9130-db.dtsi

Change-Id: Ibf4dc1d01dfdea20385fcf401cc27a9990b8c963
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/2963
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-db.dtsi | 10 ----------
 arch/arm64/boot/dts/marvell/cn9130.dtsi    |  2 +-
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
index 129bb4aad77a..82e813aa94a3 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130-db.dtsi
@@ -127,13 +127,3 @@
 	usb-phy = <&cp0_usb3_0_phy1>;
 	phy-names = "usb";
 };
-
-&cp0_pcie0 {
-	status = "okay";
-	ranges =
-		/* downstream I/O */
-		<0x81000000 0 0xf8000000 0 0xf8000000 0 0x10000
-		/* non-prefetchable memory */
-		0x82000000 0 0xc0000000 0 0xc0000000 0 0x1000000>;
-};
-
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index 9b3a834549fa..c0871d2aafca 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -27,7 +27,7 @@
 #define CP110_BASE			(CP110_BASE_OFFSET + \
 						(CP110_NUM * CP110_SPACE_SIZE))
 
-#define CP110_PCIE_MEM_SIZE		(0x1000000)
+#define CP110_PCIE_MEM_SIZE		(0xf00000)
 #define CP110_PCIE_BUS_MEM_CFG		(0x82000000)
 
 /* CP110-0 Settings */
-- 
2.17.1

